src/configure/configure.ts

```diff
@@ -0,0 +1,246 @@
+                        break;
+                    case "app,linux":
+                    case "functionapp":
+                    case "app,linux,container":
+                    default:
+                        throw new Error(`App of kind: ${azureResource.kind} is not yet supported.`);
+                }
+                break;
+            default:
+                throw new Error(`Resource of type: ${azureResource.type} is not yet supported for configuring pipelines.`);
+        }
+    }
+
+    // also check if the node type is of  file explorer type and extra the git repo details in that case.
+}
+
+async function getSourceRepositoryDetails(): Promise<void> {
+    let sourceOptions: Array<QuickPickItem> = [{ label: SourceOptions.BrowseLocalMachine }];
+    if (vscode.workspace && vscode.workspace.rootPath) {
+        sourceOptions.push({ label: SourceOptions.CurrentWorkspace });
+    }
+
+    let selectedSourceOption = await extensionVariables.uiExtensionVariables.ui.showQuickPick(
+        sourceOptions,
+        { placeHolder: "Select the folder or repository to deploy" }
+    );
+
+    let workspacePath = "";
+    switch (selectedSourceOption.label) {
+        case SourceOptions.BrowseLocalMachine:
+            let selectedFolder: vscode.Uri[] = await vscode.window.showOpenDialog(
+                {
+                    openLabel: "Select the git repository folder to deploy",
+                    canSelectFiles: false,
+                    canSelectFolders: true,
+                    canSelectMany: false
+                }
+            );
+            if (selectedFolder && selectedFolder.length > 0) {
+                workspacePath = selectedFolder[0].fsPath;
+            }
+            break;
+        case SourceOptions.CurrentWorkspace:
+        default:
+            workspacePath = vscode.workspace.rootPath;
+    }
+
+    extensionVariables.inputs.sourceRepositoryDetails = await extensionVariables.sourceRepositoryService.getGitRepoDetails(workspacePath);
+    extensionVariables.inputs.sourceRepositoryDetails.localPath = workspacePath;
+
+    if (extensionVariables.inputs.sourceRepositoryDetails.sourceProvider === SourceProviderType.AzureRepos) {
+        extensionVariables.inputs.sourceRepositoryDetails.repositoryId = await extensionVariables.azureDevOpsService.getRepositoryId(extensionVariables.inputs.sourceRepositoryDetails.repositoryName, extensionVariables.inputs.sourceRepositoryDetails.remoteUrl);
+    }
+}
+
+async function getAzureDevOpsDetails(): Promise<void> {
+    // TODO: handle space in project name and repo name.
+    if (!extensionVariables.azureDevOpsService.getOrganizationName()) {
+        let organizationList: string[] = await extensionVariables.azureDevOpsService.listOrganizations();
+        let selectedOrganization = await extensionVariables.uiExtensionVariables.ui.showQuickPick(organizationList.map((org) => { return { label: org }; }), { placeHolder: "Select Azure DevOps Organization" });
```

Comment:
why do we need to do extensionVariables.uiExtensionVariables.ui.showQuickPick instead of simply vscode.showQuickPick ?

Reply:
_vscode-azureextensionui_ has implemented certain more features in ShowQuickPick, such as 

- Saving state 

- Showing recently used at the top

- Not closing on focus shift
Hence leveraged these features using: extensionVariables.uiExtensionVariables.ui.showQuickPick

If we use vscode.showQuickPick, we need to add these features. As we were already using _vscode-azureextensionui_, it was easy to reuse their functions.

Reply:
got it.